src/TPProcessor.cpp

```diff
@@ -104,27 +108,52 @@ TPProcessor::conf(const appmodel::DataHandlerModule* conf)
 //   //ci.add(info);
 // }
 
+void
+TPProcessor::generate_opmon_data()
+{
+  opmon::TPProcessorInfo info;
+
+  info.set_tp_received_count( m_tp_received_count.load() );
+  info.set_ta_made_count( m_ta_made_count.load() );
+  info.set_ta_sent_count( m_ta_sent_count.load() );
+  info.set_ta_failed_sent_count( m_ta_failed_sent_count.load() );
+
+  this->publish(std::move(info));
+}
 
 /**
  * Pipeline Stage 2.: Do software TPG
  * */
 void
 TPProcessor::find_ta(const TriggerPrimitiveTypeAdapter* tp,  std::shared_ptr<triggeralgs::TriggerActivityMaker> taa)
 {
-	
+  m_tp_received_count++;	
   std::vector<triggeralgs::TriggerActivity> tas;
   taa->operator()(tp->tp, tas);
 
   while (tas.size()) {
+      m_ta_made_count++;
       if (!m_ta_sink->try_send(std::move(tas.back()), iomanager::Sender::s_no_block)) {
         ers::warning(TADropped(ERS_HERE, tp->tp.time_start, m_sourceid.id));
-        m_tas_dropped++;
+        m_ta_failed_sent_count++;
       }
-      m_new_tas++;
+      m_ta_sent_count++;
```

Comment:
```suggestion
    else {
        m_tc_sent_count++;
    }
```
Ditto re. `sent_count` currently being iterated even if TA was not sent. The else above should work.

Reply:
The `continue` suggested elsewhere in this case should not be considered, because `tas.pop_back()` wouldn't be triggered.

Reply:
Artur the changes you requested to be checked from Giovanna are ok. This is the right way to do it. Even readout modules was changed in that way.